udata/tests/test_activity.py

```diff
@@ -52,29 +52,30 @@ def test_create_and_retrieve_for_related(self):
         for activity in activities:
             self.assertIsInstance(activity, FakeActivity)
 
-    def check_emitted(self, sender, activity):
+    def callback(self, sender, activity):
         self.assertEqual(sender, FakeActivity)
         self.assertIsInstance(activity, FakeActivity)
-        self.emitted = True
+        self.callback_called = True
 
-    def test_emit_signal(self):
-        '''It should emit a signal on new activity'''
+    def test_callback_registration(self):
+        '''Check that the registered callback is executed when a new activity
+        is saved'''
         self.emitted = False
-        with FakeActivity.on_new.connected_to(self.check_emitted):
+        with FakeActivity.on_new.connected_to(self.callback, sender=FakeActivity):
             FakeActivity.objects.create(actor=self.user, related_to=self.fake)
 
-        self.assertTrue(self.emitted)
+        self.assertTrue(self.callback_called)
 
-    def test_class_shortcut(self):
-        '''It should emit a signal on new activity'''
+    def test_emit(self):
```

Comment:
I think the method_name was good, it was meant to test the `connect()` class method. But you're right, the `emit()` class method wasn't tested. So I think we should have 2 tests.